### PR TITLE
Added-Real-Time-Face-Detection-Function

### LINKED ISSUE #1: Implement Real-time Face Recognition in main.py
### Description
In the current project, we have identified the need to implement real-time face recognition functionality in the `main.py` file. This feature would enable the system to perform face recognition on video frames captured from a camera source and display the results in real time.

### Expected behavior
- The real_time_detection function in `main.py` should capture video from a camera source.
- For each frame, the system should perform face recognition using the trained model.
- The recognized faces should be annotated on the frame.
- The user should be able to press 'q' to quit the real-time detection.

### Additional context

- Assume that the code for face recognition and model loading is already implemented in the project.
- Comments indicating the structure and functionality of the real_time_detection function are present in the code.
- Refer to `# TODO 1`  in main.py to locate the function you need to modify.

### PR DESCRIPTION
**Pull request by meenakshi3151**
## Description

In the main.py file, I have added the code in the real-time-detection which  captures the video frame and calls for the recognize face function which will recognize the face using trained data. If the person is not in the trained data then he/she will be annotated as Unknown .

## Related Issue

Fixes #1 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Motivation and Context

This change will make the system more dynamic by capturing the real time video.

## How Has This Been Tested?

This has been tested by showing different images to the camera which are in the trained model and it is able to annotate the images correctly .

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

**Comment by sanskaromar** [comment]
What's with this PR name? 
Additionally, Get issue reassigned to you from Contrihub and make a PR request there.

Update this `Fixes # (issue)` with the issue number this PR fixes.

File: main.py [hunk 1]
```diff
@@ -62,6 +121,7 @@ def main():
 
             if sub_choice == '1':
                 real_time_detection(security_system)
+              
             elif sub_choice == '2':
```
Comment (sanskaromar) [inline main.py]: Why this newline here?

File: main.py [hunk 2]
```diff
@@ -8,20 +8,79 @@ def train_new_model():
...
+    return faces
+
+
 def real_time_detection(security_system):
-    # TODO 1
-    # Capture video from a camera (you can adjust the video source)
     
+    capture_vid = cv2.VideoCapture(0)  
+    # Now perform face recognition on the frame
+    #Now while the condition is true , we will capture the image
```
Comment (sanskaromar) [inline main.py]: Inconsistent spacing in comments.

File: main.py [hunk 3]
```diff
@@ -8,20 +8,79 @@ def train_new_model():
...
+        ret, frame = capture_vid.read()
         # Perform face recognition on the frame
+        recognized_faces = recognize_face_dummy(frame)
 
+       
+        for (x, y, w, h) in recognized_faces:
+            cv2.rectangle(frame, (x, y), (x + w, y + h), (0, 255, 0), 2)  
+        cv2.imshow('Real-time Face Recognition', frame)
+        if cv2.waitKey(1) & 0xFF == ord("q"):
+          break
```
Comment (sanskaromar) [inline main.py]: There was no need to implement all this. This is already implemented in `app/security_system.py`. Use face recognizer from there.

File: main.py [hunk 4]
```diff
@@ -8,20 +8,79 @@ def train_new_model():
...
         # Press 'q' to quit
+       
+
```
Comment (sanskaromar) [inline main.py]: Where is this functionality?

File: main.py [hunk 5]
```diff
@@ -8,20 +8,79 @@ def train_new_model():
...
+         if confidence < 50:
+            recognized_label = label_to_name[label]
+         else:
+             recognized_label = "Unknown"
+         cv2.rectangle(frame, (x, y), (x + w, y + h), (0, 255, 0), 4)
+
+         cv2.putText(frame, recognized_label, (x, y-10), cv2.FONT_HERSHEY_SIMPLEX, 0.9, (255, 100, 0), 2)
+    return faces
+
+
```
Comment (sanskaromar) [inline main.py]: You did all this extra work when there was no need for this. 😔

**It's ALWAYS a great idea to read README and documentation if they are given.**

File: main.py [hunk 6]
```diff
@@ -8,20 +8,79 @@ def train_new_model():
     selected_algorithm = input("Select the algorithm for training (LBPH/Eigen/Fisher): ").strip()
     train_face_recognition_model(selected_algorithm)
 
+face_classifier = cv2.CascadeClassifier(
+    cv2.data.haarcascades + "haarcascade_frontalface_default.xml"
+)
+
```
Comment (sanskaromar) [inline main.py]: :(

**Review by sanskaromar — CHANGES_REQUESTED** [review]
Please make required changes.

File: main.py (hunk 4 above)
Comment (meenakshi3151) [reply main.py]: Sir line 75 and 76 contains this functionality

File: main.py (hunk 5 above)
Comment (meenakshi3151) [reply main.py]: Ok sir

File: main.py (hunk 4 above)
Comment (sanskaromar) [reply main.py]: Okay. Then write this comment there.

**Comment by sanskaromar** [comment]
Please resolve conflicts in `security_system.py`.

**Pull request closed without merging**